contracts/core/OriginCore.sol

```diff
@@ -353,9 +372,20 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
             _s,
             verifiedWeight,
             MetaBlock.requiredWeightForSuperMajority(
-                    stake.totalWeightAtHeight(height)
+                stake.totalWeightAtHeight(height)
             )
         );
+
+        /* Meta-block should only be committed once. */
+        if (_kernelHash == openKernelHash) {
+            commitMetaBlock(
+                _kernelHash,
+                _transitionHash,
+                height,
+                verifiedWeight
+            );
+        }
```

Comment:
I don't understand why the meta block is committed if merely the hash of the kernel is correct. Shouldn't it depend on the weight that was verified?